network/wifi/at_parser.c

```diff
@@ -204,13 +184,52 @@ struct at_desc {
 	}			ipd_stat;
 	/* Variable where current message length is built */
 	uint32_t		ipd_len;
+	/* Will be called when a new connection is created or closed */
+	void			(*connection_callback)(void *ctx, enum at_event,
+			uint32_t conn_id, struct circular_buffer **cb);
+	/* Context that will be passed to the callback */
+	void			*callback_ctx;
 };
 
 /******************************************************************************/
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
-static void set_params(struct at_buff *dest, uint8_t *fmt, ...);
+/* Call connection_callback with the specified event.
+   With this callback, an upper layer can organize its connection structures and
+   buffers based on the events that occurs on each connection id */
+static void dispatch_event(struct at_desc *desc, enum at_event event)
+{
+	struct connection_desc *conn;
+
+	conn = &desc->conn[desc->current_conn];
+
+	if (event == AT_NEW_CONNECTION) {
+		if (!conn->active) {
+			/* Notify that a new connection is started */
+			desc->connection_callback(desc->callback_ctx, event,
+						  desc->current_conn,
+						  &conn->cbuff);
+			/* If user sets cbuff -> connection is set as active */
+			if (conn->cbuff)
+				conn->active = true;
+			//else
+				// This could be done only if implement
+				//at_run_cmd with uart_write_nonblocking
```

Comment:
```
/* This could be done only if implement
 * at_run_cmd with uart_write_nonblocking
 * union in_out_param param;
 * param.in.conn_id = desc->current_conn;
 * at_run_cmd(desc, AT_STOP_CONNECTION,
 * AT_SET_OP, &param); 
 */
```
Please use this style of commenting.

Reply:
fixed